### ISSUE TITLE
Handle dune files in vendored packages

### ISSUE THREAD

**Issue by Leonidas-from-XIV** [tarides/opam-monorepo#370]
Menhir ships with `vendored_pprint` (and `vendored_fix`) which are used in the rest of Menhir, but at the moment these get renamed, while the reference to them does not, so the build fails as the Menhir `dune` files refer to `vendored_pprint` while the public_name is `vendored_pprint.<hash>` and only the non-public name is `vendored_pprint` (but that name is not exposed outside of  the `vendored_pprint` package).

TL;DR: opam-monorepo changes references from `public_names` to private names, but that rename operation is not always valid.
